src/components/NavBar/index.tsx

```diff
@@ -1,50 +1,39 @@
-import { useState } from 'react'
+import { useEffect, useState } from 'react'
 import { useRouter } from 'next/router'
-import {
-  ChatAltIcon,
-  ExclamationCircleIcon,
-  HomeIcon,
-  UserIcon
-} from '@heroicons/react/outline'
+import { ChatAltIcon, HomeIcon } from '@heroicons/react/outline'
 
 import { NavIcon } from '@/components/NavBar/navicon'
 import NavLink from '@/components/NavBar/navlink'
 
 export default function NavBar() {
   // Gets top level path, capitalises first letter
   const { asPath } = useRouter()
-  const pathStart = asPath.split('/')[1]
-  const currentPageTitle =
-    pathStart === ''
-      ? 'Home'
-      : `${pathStart.charAt(0).toUpperCase()}${pathStart
-          .slice(1)
-          .toLowerCase()}`
 
-  const [currentPage, setCurrentPage] = useState(currentPageTitle)
+  const [currentPage, setCurrentPage] = useState('')
   const iconClasses = 'inline-block h-7 w-8 hover:text-primary'
 
+  useEffect(() => {
+    const pathStart = asPath.split('/')[1]
+    const currentPageTitle =
+      pathStart === ''
+        ? 'Home'
+        : `${pathStart.charAt(0).toUpperCase()}${pathStart
+            .slice(1)
+            .toLowerCase()}`
+    setCurrentPage(currentPageTitle)
+  }, [asPath])
+
```

Comment:
I don't think this refactor is necessary as setCurrentPage is already called when a navlink is clicked (onMouseDown property). This code causes NavBar to be re-rendered one additional time

Reply:
good catch arun 👍

Reply:
Thanks for testing this out for me